### PR TITLE
Update the code in manual and examples to be formatted more nicely

### PR DESCRIPTION
**Pull request by eddiebergman**
Formats the code in the docs as it would be written and lint acceptable.

### CODE REVIEW

File: examples/40_advanced/example_pandas_train_test.py [hunk 1]
```diff
@@ -118,26 +118,30 @@ def get_runhistory_models_performance(automl):
...
+best_values = pd.Series({
+    'ensemble_optimization_score': -np.inf,
+    'ensemble_test_score': -np.inf
+})
 for idx in ensemble_performance_frame.index:
-    if (
-        ensemble_performance_frame.loc[idx, 'ensemble_optimization_score']
-        > best_values['ensemble_optimization_score']
-    ):
+    ensemble_score = ensemble_performance_frame.loc[idx, 'ensemble_optimization_score']
```
Comment (mfeurer) [inline examples/40_advanced/example_pandas_train_test.py:127]: I disagree on this change - isn't the former one easier to read and also preferable as it doesn't create an intermediate variable?

File: examples/40_advanced/example_pandas_train_test.py [hunk 2]
```diff
@@ -118,26 +118,30 @@ def get_runhistory_models_performance(automl):
...
+    'single_best_optimization_score': -np.inf,
+    'single_best_test_score': -np.inf,
+    'single_best_train_score': -np.inf
+})
 for idx in individual_performance_frame.index:
-    if (
-        individual_performance_frame.loc[idx, 'single_best_optimization_score']
-        > best_values['single_best_optimization_score']
-    ):
+    single_best_score = individual_performance_frame.loc[idx, 'single_best_optimization_score']
```
Comment (mfeurer) [inline examples/40_advanced/example_pandas_train_test.py:141]: Same here.

File: examples/40_advanced/example_metrics.py [hunk 3]
```diff
@@ -79,9 +79,9 @@ def error_wk(solution, prediction, dummy):
 cls.fit(X_train, y_train)
 
 predictions = cls.predict(X_test)
-print("Accuracy score {:g} using {:s}".
-      format(sklearn.metrics.accuracy_score(y_test, predictions),
-             cls.automl_._metric.name))
+score = sklearn.metrics.accuracy_score(y_test, predictions)
```
Comment (mfeurer) [inline examples/40_advanced/example_metrics.py:82]: Do we need these intermediate metrics? Isn't the idea of f-strings to circumvent them?

File: examples/80_extending/example_extending_regression.py [hunk 4]
```diff
@@ -48,8 +49,7 @@ def fit(self, X, y):
                                                           kernel=self.kernel,
                                                           gamma=self.gamma,
                                                           degree=self.degree,
-                                                          coef0=self.coef0,
-                                                          )
+                                                          coef0=self.coef0)
```
Comment (mfeurer) [inline examples/80_extending/example_extending_regression.py:52]: Shouldn't there be a linebreak after the bracket after KernelRidge?

File: doc/manual.rst [hunk 5]
```diff
@@ -71,16 +74,18 @@ Turning off preprocessing
 Preprocessing in *auto-sklearn* is divided into data preprocessing and
 feature preprocessing. Data preprocessing includes One-Hot encoding of
 categorical features, imputation of missing values and the normalization of
-features or samples. These steps currently cannot be turned off. Feature
-preprocessing is a single transformer which implements for example feature
+features or samples. Dataprerocessing steps cannot be turned off as this ensures
```
Comment (mfeurer) [inline doc/manual.rst:77]: Wasn't that a feature that was implemented by @rabsr in #977? Maybe we need to extend the documentation on this?

File: examples/40_advanced/example_pandas_train_test.py (hunk 1 above)
Comment (eddiebergman) [reply examples/40_advanced/example_pandas_train_test.py:127]: I just found it very crowded personally but I'm okay with either. I'll change it back

File: examples/40_advanced/example_pandas_train_test.py (hunk 2 above)
Comment (eddiebergman) [reply examples/40_advanced/example_pandas_train_test.py:141]: > I just found it very crowded personally but I'm okay with either. I'll change it back

File: examples/40_advanced/example_metrics.py (hunk 3 above)
Comment (eddiebergman) [reply examples/40_advanced/example_metrics.py:82]: Just for legibility sake I thought it better to have as intermediate vars. 

2 of the examples were actually using the wrong scorer as I imagine it was embedded into the long print expression.
I will comment on the diff where it occured

File: examples/80_extending/example_extending_regression.py (hunk 4 above)
Comment (eddiebergman) [reply examples/80_extending/example_extending_regression.py:52]: I'm not sure what you mean by this
* Do you mean just a linebreak between the two expressions `self.estimator = ...` and `self.estimator.fit(...)`?
```
self.estimator = sklearn.kernel_ridge.KernelRidge(alpha=self.alpha,
                                                  kernel=self.kernel,
                                                  gamma=self.gamma,
                                                  degree=self.degree,
                                                  coef0=self.coef0)
                                                  
self.estimator.fit(X, y)
return self
```
* You prefer the linespace by having the hanging bracket as it was?
```
self.estimator = sklearn.kernel_ridge.KernelRidge(alpha=self.alpha,
                                                  kernel=self.kernel,
                                                  gamma=self.gamma,
                                                  degree=self.degree,
                                                  coef0=self.coef0,
                                                  )
self.estimator.fit(X, y)
return self
```

File: doc/manual.rst (hunk 5 above)
Comment (eddiebergman) [reply doc/manual.rst:77]: Yes it needs to be documented properly, this was just a quick fix to give some space and clearer reference to what's being talked about rather than 'these'.

I believe this should be properly handled in the issue for addressing the walkthrough manual

File: examples/40_advanced/example_metrics.py [hunk 6]
```diff
@@ -106,9 +106,13 @@ def error_wk(solution, prediction, dummy):
 cls.fit(X_train, y_train)
 
 predictions = cls.predict(X_test)
-print("Accuracy score {:g} using {:s}".
-      format(sklearn.metrics.accuracy_score(y_test, predictions),
-             cls.automl_._metric.name))
```
Comment (eddiebergman) [inline examples/40_advanced/example_metrics.py:111]: Just an example of where the example was incorrect, hence me wanting to split it out and make it more clear. In general my preference is to not have any calculating functions performed in a formatted string as it can easily brushed over.
However `cls._automl_._metric.name` could definitely go into the string, there probably wasn't much point to take it out other than just to show how to get the metric name as a single expression.

File: examples/40_advanced/example_pandas_train_test.py [hunk 7]
```diff
@@ -118,26 +118,30 @@ def get_runhistory_models_performance(automl):
...
 individual_performance_frame = get_runhistory_models_performance(cls)
-best_values = pd.Series({'single_best_optimization_score': -np.inf,
-                         'single_best_test_score': -np.inf,
-                         'single_best_train_score': -np.inf})
+best_values = pd.Series({
+    'single_best_optimization_score': -np.inf,
+    'single_best_test_score': -np.inf,
+    'single_best_train_score': -np.inf
+})
 for idx in individual_performance_frame.index:
```
Comment (eddiebergman) [inline examples/40_advanced/example_pandas_train_test.py:140]: Just a note on why I would prefer these if curious. Any editor with tab indicators will have these line up nicely when beginning on a new line as they are lined up as a tab. Contrasted with the hanging version where they are lined up to the previous parameters and not to a tabline may not necessarily be on a tabline.

Really not important but you'll notice me use this a lot more often if possible.

File: examples/80_extending/example_extending_regression.py (hunk 4 above)
Comment (mfeurer) [reply examples/80_extending/example_extending_regression.py:52]: I mean:
```
self.estimator = sklearn.kernel_ridge.KernelRidge(
    alpha=self.alpha,
    kernel=self.kernel,
    gamma=self.gamma,
    degree=self.degree,
    coef0=self.coef0,
)
                                                  
self.estimator.fit(X, y)
return self
```

File: examples/40_advanced/example_metrics.py (hunk 6 above)
Comment (mfeurer) [reply examples/40_advanced/example_metrics.py:111]: Alright.

File: examples/40_advanced/example_metrics.py (hunk 3 above)
Comment (mfeurer) [reply examples/40_advanced/example_metrics.py:82]: Alright.

File: examples/40_advanced/example_pandas_train_test.py (hunk 7 above)
Comment (mfeurer) [reply examples/40_advanced/example_pandas_train_test.py:140]: I fully agree that this looks a lot nicer.

**Review by mfeurer — APPROVED** [review]
